frontend/src/components/modals/EditCoach.tsx

```diff
@@ -1,36 +1,42 @@
 import React from "react";
-import { useEditUserMutation } from "../../features/user/apiSlice";
+import { useEditCoachMutation } from "../../features/user/apiSlice";
 import { useForm } from "react-hook-form";
 import ModalLayout from "./ModalLayout";
 import Alert from "../ui/Alert";
 import InputField from "../ui/InputField";
 import Button from "../ui/Button";
+import Loader from "../ui/Loader";
 
-const EditUser = ({
+const EditCoach = ({
   closePopup,
   jwt,
-  user,
-  id,
+  coach,
 }: {
   closePopup: () => void;
   jwt: string;
-  id: any;
```

Comment:
Avoid `any` unless it has been complicated too much. for example, you can add it to` id` and if you find it possible and easy, write a type of coach and user outside and reference their type here.